### ISSUE TITLE
[java] UnusedPrivateField: Do not flag fields annotated with @Version

### ISSUE THREAD

**Issue by numeralnathan** [pmd/pmd#3824]
**Affects PMD Version:** 6.43.0

**Rule:** [UnusedPrivateField](https://pmd.github.io/pmd-6.43.0/pmd_rules_java_bestpractices.html#unusedprivatefield)

**Description:**

Optimistic locking for JPA requires a field annotated with `@javax.persistence.Version`.  These fields rarely have a getter or setter.  Please fix `UnusedPrivateField` to ignore fields annotated with `@javax.persistence.Version`.

**Code Sample demonstrating the issue:**

```
@Entity
@Table(name = "my_table")
public class MyTable implements Serializable
{
   private static final long serialVersionUID = 1L;

   @Id
   @Column(name = "id", nullable = false)
   private long m_id;

   // other fields

   @Version
   @Column(name = "optimistic_lock", nullable = false)
   private short m_optimisticLock;    // PMD Violation: UnusedPrivateField

   /*
    * For JPA
    */
   public MyTable()
   {
      // nothing to do
   }

   public long getId()
   {
      return m_id;
   }

   // getters and setters for other fields except m_optimisticLock
}
```

**Expected outcome:**

PMD reports a violation at line `private short m_optimisticLock;    // PMD Violation: UnusedPrivateField`, but that's wrong. That's a false positive.

**Running PMD through:** Eclipse PMD Plugin

**Comment by pzygielo** [comment]
- https://github.com/pmd/pmd/issues/3825#issuecomment-1059802085
